### PR TITLE
[SKIP CI] .github: add zephyr build

### PR DESCRIPTION
**Pull request by marc-hb**
3 commits, see messages, especially: `xtensa-build-zephyr: do not clone a second version of sof.git`

Related rimage.h background info: https://github.com/zephyrproject-rtos/sof/pull/7#issuecomment-814539846

### CODE REVIEW

**Comment by marc-hb** [comment]
The checktree failure in https://github.com/thesofproject/sof/pull/4114/checks?check_run_id=2462963981 has been introduced by #4111, it's unrelated

**Comment by cujomalainey** [comment]
LGTM to me, but wont approve as my bash is sub par

File: zephyr/docker-build.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+set -e
+set -x
+
+# export http_proxy=...
+# export https_proxy=...
+
+unset ZEPHYR_BASE
+
+sudo --preserve-env=http_proxy,https_proxy apt-get update
+sudo --preserve-env=http_proxy,https_proxy apt-get -y install libssl-dev
```
Comment (marc-hb) [inline zephyr/docker-build.sh:17]: Submitted libssl-dev addition to the docker image in https://github.com/zephyrproject-rtos/docker-image/pull/58

**Comment by marc-hb** [comment]
> LGTM to me, but wont approve as my bash is sub par

While not always easy to write, I believe all shell scripts should be mostly readable even by beginners. For every cryptic shell "trick" there's hopefully a comment next to it. I hope you would have asked if anything had confused you. 

PS: shellcheck is an outstanding teacher. I mean for a program.

**Comment by dbaluta** [comment]
> The checktree failure in https://github.com/thesofproject/sof/pull/4114/checks?check_run_id=2462963981 has been introduced by #4111, it's unrelated

This should be fixed with: https://github.com/thesofproject/sof/pull/4116

**Review by lgirdwood — APPROVED** [review]
Thanks @marc-hb this was really needed.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by cujomalainey** [comment]
> > LGTM to me, but wont approve as my bash is sub par
> 
> While not always easy to write, I believe all shell scripts should be mostly readable even by beginners. For every cryptic shell "trick" there's hopefully a comment next to it. I hope you would have asked if anything had confused you.
> 
> PS: shellcheck is an outstanding teacher. I mean for a program.

Yes i usually get the gist, ive read most of the way through my linux phrase book which is all bashism's for the most part, but I study and practice are two very different levels of knowledge :)

**Comment by marc-hb** [comment]
> SOFCI TEST

I believe this only works for jenkins which I disabled with `[SKIP CI]`. Sometimes it takes many force-pushes to get some `.github/workflows/` right and that's basically a DoS attack on Jenkins. Jenkins does not use this build script (yet?). I force-pushed an empty `git commit --amend` instead, that just cleared the checktree error that has been fixed.

 -----

Q&A related to this PR: https://github.com/zephyrproject-rtos/zephyr/discussions/34713 `Continuous Integration with multiple git repos? (using west or similar)`

**Comment by lgirdwood** [comment]
Thanks @marc-hb.

**Pull request merged**